pkg/rpaas/client/client.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2019 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type RpaasClient struct {
+	hostAPI      string
+	tsuruTarget  string
+	tsuruToken   string
+	target       string
+	tsuruService string
+	httpClient   *http.Client
+}
+
+func New(hostAPI string) *RpaasClient {
+	client := &RpaasClient{httpClient: &http.Client{}, hostAPI: hostAPI}
+	return client
+}
+
+func NewTsuruClient(tsuruAPI, service, token string) (*RpaasClient, error) {
+	if tsuruAPI == "" || token == "" || service == "" {
+		return nil, fmt.Errorf("cannot create client without either tsuru target, token or service")
+	}
+	return &RpaasClient{
+		httpClient:   &http.Client{},
+		tsuruTarget:  tsuruAPI,
+		tsuruToken:   token,
+		tsuruService: service,
+	}, nil
+}
+
+type flavor struct {
```

Comment:
We should export these structs. By the way, we can move them to a types only package, what do you think?

Reply:
sure, client.go is already getting big

Reply:
i added type/types.go but am using a types.go inside the client package until I learn how to import packages properly in go LOL, apparently I should have put the porject inside my ${GOPATH}